src/main/scala/mesosphere/marathon/api/v2/json/Formats.scala

```diff
@@ -123,9 +123,10 @@ trait Formats
         "startedAt" -> launched.status.startedAt,
         "stagedAt" -> launched.status.stagedAt,
         "ports" -> launched.hostPorts,
-        "version" -> launched.appVersion,
-        "ipAddresses" -> launched.ipAddresses
-      )
+        "version" -> launched.appVersion
+      ) ++ launched.ipAddresses.map { ipAddresses =>
+          Json.obj("ipAddresses" -> ipAddresses)
+        }.getOrElse(Json.obj())
```

Comment:
I think that you do not have to make it that complicated. `"ipAddresses" -> launched.ipAddresses` should suffice.

```scala
scala> case class WithOption(test: Option[String])
defined class WithOption

scala> implicit val format = play.api.libs.json.Json.writes[WithOption]
format: play.api.libs.json.OWrites[WithOption] = play.api.libs.json.OWrites$$anon$2@47fc5ae4

scala> play.api.libs.json.Json.toJson(WithOption(None))
res1: play.api.libs.json.JsValue = {}

scala> play.api.libs.json.Json.toJson(WithOption(Some("test")))
res2: play.api.libs.json.JsValue = {"test":"test"}
```